MdeModulePkg/Core/PiSmmCore/PiSmmIpl.c

```diff
@@ -667,6 +669,48 @@ SmmCommunicationCommunicate (
   return (Status == EFI_SUCCESS) ? EFI_SUCCESS : EFI_NOT_FOUND;
 }
 
+/**
+  Communicates with a registered handler.
+
+  This function provides a service to send and receive messages from a registered
+  UEFI service.  This function is part of the SMM Communication Protocol that may
+  be called in physical mode prior to SetVirtualAddressMap() and in virtual mode
+  after SetVirtualAddressMap().
+
+  @param[in] This                The EFI_SMM_COMMUNICATION_PROTOCOL instance.
+  @param[in, out] CommBuffer     A pointer to the buffer to convey into SMRAM.
+  @param[in, out] CommSize       The size of the data buffer being passed in. On exit, the size of data
+                                 being returned. Zero if the handler does not wish to reply with any data.
+                                 This parameter is optional and may be NULL.
+
+  @retval EFI_SUCCESS            The message was successfully posted.
+  @retval EFI_INVALID_PARAMETER  The CommBuffer was NULL.
+  @retval EFI_BAD_BUFFER_SIZE    The buffer is too large for the MM implementation.
+                                 If this error is returned, the MessageLength field
+                                 in the CommBuffer header or the integer pointed by
+                                 CommSize, are updated to reflect the maximum payload
+                                 size the implementation can accommodate.
+  @retval EFI_ACCESS_DENIED      The CommunicateBuffer parameter or CommSize parameter,
+                                 if not omitted, are in address range that cannot be
+                                 accessed by the MM environment.
+
+**/
+EFI_STATUS
+EFIAPI
+SmmCommunicationCommunicate (
+  IN CONST EFI_SMM_COMMUNICATION_PROTOCOL  *This,
+  IN OUT VOID                              *CommBuffer,
+  IN OUT UINTN                             *CommSize OPTIONAL
+  )
+{
+  return SmmCommunicationCommunicateCommon (
+            &mSmmCommunication,
+            CommBuffer,
+            CommBuffer,

```

Comment:
This does not resolve this issue.  CommBuffer is always a physical address.  By passing the physical address both the physical and virtual address, the use of the physical address to check the V3 GUID is still used.

Reply:
Yup.. I just replied with an email. This change will not resolve the issue. I do have one proposal through email. Please let me know how you think about it.